### PR TITLE
Add --show-password flag to user reset-password

### PR DESCRIPTION
**Pull request by dsXLII**
First go at a PR to address https://github.com/wp-cli/entity-command/issues/342 .

### CODE REVIEW

File: features/user-reset-password.feature [hunk 1]
```diff
@@ -41,3 +41,23 @@ Feature: Reset passwords for one or more WordPress users.
...
       """
+
+@require-wp-4.3
+Scenario: Reset the password of a WordPress user, and show the new password
+  Given a WP installation
+
+  When I run `wp user get 1 --field=user_pass`
+  Then save STDOUT as {ORIGINAL_PASSWORD}
+
+  When I run `wp user reset-password 1 --skip-email --show-password`
```
Comment (dsXLII) [inline features/user-reset-password.feature:52]: This absolutely should capture the new password, and the next step should check for it, but I have no idea how to do that :(

**Review by danielbachhuber — COMMENTED** [review]
Great start!

**Comment by danielbachhuber** [comment]
@dsXLII Still planning to finish this up?

**Comment by dsXLII** [comment]
I'm still here, just distracted by the day job :/

Question about porcelain. For the new app password example, since you can only create one app password at a time the output is unambiguous. But here, since you can reset multiple users' passwords at the same time, it's less clear. (If I do a `wp user reset-password 1 2 3 4` four passwords would be output, but it's not obvious that the passwords go in the same order as the user arguments.) What's the best way to handle that? Output them in that order, and document it? Don't allow --porcelain for multiple user arguments?

**Comment by danielbachhuber** [comment]
> I'm still here, just distracted by the day job :/

No worries! 

> Question about porcelain.

Good question! You are correct: we don't yet have prior art within WP-CLI.

Let's follow `git status` and output each password on a new line:

```
$ git status --porcelain
?? test1
?? test2
```

File: features/user-reset-password.feature [hunk 2]
```diff
@@ -41,3 +41,40 @@ Feature: Reset passwords for one or more WordPress users.
...
+
+  @require-wp-4.3
+  Scenario: Reset the password of a WordPress user, and show only the new password
+    Given a WP installation
+
+    When I run `wp user get 1 --field=user_pass`
+    Then save STDOUT as {ORIGINAL_PASSWORD}
+
+    When I run `wp user reset-password 1 --skip-email --porcelain`
+    Then save STDOUT as {NEW_PASSWORD}
```
Comment (dsXLII) [inline features/user-reset-password.feature:73]: TODO: Learn enough behat to have this check the output (it should be a password, currently "exactly 12 characters, no whitespace, etc")

File: src/User_Command.php [hunk 3]
```diff
@@ -1104,19 +1107,24 @@ public function import_csv( $args, $assoc_args ) {
...
 	public function reset_password( $args, $assoc_args ) {
 		$skip_email = Utils\get_flag_value( $assoc_args, 'skip-email' );
+		$show_new_pass = Utils\get_flag_value( $assoc_args, 'show-password' );
 		if ( $skip_email ) {
 			add_filter( 'send_password_change_email', '__return_false' );
 		}
 		$fetcher = new UserFetcher();
 		$users   = $fetcher->get_many( $args );
 		foreach ( $users as $user ) {
+			$new_pass = wp_generate_password();
```
[the remark this answers is not in the document]
Comment (dsXLII) [reply src/User_Command.php:1126]: Created a separate PR. :)

File: features/user-reset-password.feature (hunk 2 above)
Comment (danielbachhuber) [reply features/user-reset-password.feature:73]: The check doesn't need to be super accurate. In other cases, we simply confirm that `STDOUT` is not empty.

Behat probably doesn't support checking "exactly 12 characters, no whitespace, etc". In order to perform such a check, you'd need to write a standalone PHP file to execute the check, pass the password to the PHP file, and then have the PHP file exit with 0 or 1 depending on whether it passed or failed. But, doing so is probably overkill for this particular test.

File: features/user-reset-password.feature (hunk 1 above)
Comment (dsXLII) [reply features/user-reset-password.feature]: ... I have no idea what I was hung up on, disregard :)

**Comment by dsmith4-godaddy** [comment]
I _think_ this is ready for a more thorough review. (Resolved the phpcs issues, and a merge conflict from the "change passwords to be 24 characters" PR.)

**Review by danielbachhuber — APPROVED** [review]
Great work on this, @dsXLII ! Sorry I missed your comment earlier.

I made some minor formatting tweaks with 6bb768194a02d3c144430012152d4f182586159d

**Pull request merged**